### DISCUSSION TITLE
Why do regular interpolated strings not allow repeating the $ like raw string literals do?

### DISCUSSION THREAD

**Discussion by BhaaLseN** [dotnet/csharplang#6935] [Q&A] [answered]
I was just preparing a few slides for my coworkers to highlight new features in C#11, and while writing up the section for raw string literals I wondered..."does this work on regular interpolations?" (Spoiler: the answer is _no_, and this discussion is to find out _why_)

Raw string literals let us do this (despite having their main use-case as multi-line literals, which are otherwise a pain to type out/escape things):
```c#
int theAnswer = 42;
string javaScriptCode = $$"""var theQuestion = { expectedAnswer: {{theAnswer}}, prompt: 'What is the answer to life, the universe and everything?' };""";
```

But regular interpolations do not allow this:
```c#
int theAnswer = 42;
string javaScriptCode = $$"var theQuestion = { expectedAnswer: {{theAnswer}}, prompt: 'What is the answer to life, the universe and everything?' };";
```
Visual Studio 2022 17.4 correctly does the code-highlighting, but brings up `error CS9004: Not enough quotes for raw string literal.` if you actually try to use it.

The error is somewhat odd as well, because it feels like the `$$` would be an interpolation feature, not a raw string literal feature (because it _doesn't_ complain about the `$$` but instead about the opening `"`). Particularly since it is mentioned at length in the [string interpolation "Structure of an interpolated string" section](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated#structure-of-an-interpolated-string) in the last few paragraphs; but only barely touched on the [raw string literal](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/raw-string) page in the last sentence.

Granted, this saves the whole lot of 4 characters (and could've been a lot uglier if I chose to name the variable `json` instead); but at this point I'm expecting questions like "Does this work with regular interpolations?" and the obvious "Why _not_?". Both are legitimate questions that I can see coming from both seasoned developers and less experienced juniors alike. Besides the less obvious side-effect that it might push certain kinds of developers to simply default to single-line raw string literals for _everything_...

So, to type it out loud: Is there any specific reason why this was disallowed?

**Comment by CyrusNajmabadi** [comment, marked as answer]
> So, to type it out loud: Is there any specific reason why this was disallowed?

There is no decision to disallow.  The language works in the opposite direction.  We decide what to allow.  There was never any proposal or suggestion to allow this, so the concept has never even come up for discussion.  :)

If you'd like this to be allowed, we can work through a proposal here.  If someone on the LDM is interested in championing it, it could happen.

That said, given the costs here (and the easy solution of just using a raw-string instead), i imagine the motivation to pick it up will be low.  

Thanks!

**Reply by BhaaLseN** [reply]
Good enough for me, thanks for replying!
